src/typesystem.rs

```diff
@@ -106,7 +106,25 @@ impl DataType {
     }
 }
 
-type VarArgs = Option<String>;
+#[derive(Debug, Clone, PartialEq)]
+pub enum VarArgs {
+    Sized(Option<String>),
```

Comment:
what is the String here? is it the type-name?
don't you alreay have the typename in the VariableIndexEntry?